buffer/src/main/java/io/netty/buffer/PoolThreadCache.java

```diff
@@ -231,11 +242,14 @@ boolean add(PoolArena<?> area, PoolChunk chunk, long handle, int normCapacity, S
      *  Should be called if the Thread that uses this cache is about to exist to release resources out of the cache
      */
     void free() {
-        ThreadDeathWatcher.unwatch(thread, freeTask);
-        free0();
+        if (freeTask != null) {
+            assert deathWatchThread != null;
+            ThreadDeathWatcher.unwatch(deathWatchThread, freeTask);
+        }
+        free0(Thread.currentThread());
     }
 
-    private void free0() {
+    private void free0(Thread thread) {
```

Comment:
do we need to take a Thread parameter here? It is a bit misleading that this may be necessary to do the free operation. Consider just using `Thread.currentThread()` inline in the log statement.